File: TodayFitComplete_Backend/src/main/java/com/SJY/TodayFitComplete_Backend/dto/response/member/MemberTokenResponse.java
package com.SJY.TodayFitComplete_Backend.dto.response.member;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.security.core.userdetails.UserDetails;

/**
 * 사용자 정보 반환 및 토큰.
 */
@Getter
@Setter
@NoArgsConstructor
public class MemberTokenResponse {
    private String email;
    private String token;

    @Builder
    public MemberTokenResponse(String email, String token) {
        this.email = email;
        this.token = token;
    }

    public static MemberTokenResponse fromEntity(UserDetails member, String token) {
        return MemberTokenResponse.builder()
                .email(member.getUsername())
                .token(token)
                .build();
    }
}

File: TodayFitComplete_Frontend/app/src/main/java/kr/ac/kumoh/ce/s20190633/todayfitcomplete_frontend/Dto/Member/MemberTokenResponse.kt
package kr.ac.kumoh.ce.s20190633.todayfitcomplete_frontend.Dto.Member

// MemberTokenResponse 클래스: 사용자 로그인 후 서버에서 발급된 토큰과 사용자 이메일을 클라이언트로 전달하는 데이터 전송 객체 (DTO)
data class MemberTokenResponse(
    val email: String,  // 사용자 이메일 주소
    val token: String   // 사용자 인증을 위한 토큰
)